### ISSUE TITLE
Create per-thing (selector, property etc) basic*Test utils

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#1313]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

This would help reduce the number of unnecessary tests, as well as the added control of possible errors.

**Comment by davidtheclark** [comment]
Can you describe what specific tests to add and why? Lots of the ways we look out for those things seem rule-specific.

**Comment by evilebottnawi** [comment]
@davidtheclark standard check ignore all variables, interpolation (maybe add more). In all rules we ignore this, why not include in basic check?

**Comment by davidtheclark** [comment]
How do we know generically which CSS constructs to check for variables or interpolation?

**Comment by evilebottnawi** [comment]
@davidtheclark or we may create `basicChecksLess`, `basicChecksScss` and include their where need?

**Comment by davidtheclark** [comment]
I'm just saying that I don't understand what these would look like. I will need to see some code to understand.

**Comment by evilebottnawi** [comment]
@davidtheclark 
basicSelectorsChecksScss
```js
// These should pass for *almost* every rule
export default [
  {
    code: "#{$variable} { }",
    description: "empty stylesheet",
  },
  {
    code: "foo#{$variable} { }",
    description: "empty stylesheet",
  },
  {
    code: "&-foo { }",
    description: "empty stylesheet",
  }
  // and etc
]
```
Also for unit, at-rule and etc?

**Comment by evilebottnawi** [comment]
It would save time on testing.

**Comment by davidtheclark** [comment]
Only rules that check selectors should bother with this, though, right? Which is not the majority of rules.

Maybe we should just conventionalize another way to reuse common tests like this. We could create objects in `testUtils/` that tests could import and use?

**Comment by jeddy3** [comment]
> We could create objects in testUtils/ that tests could import and use?

This sgtm. 

I also agree that breaking them down by *thing* would be prudent e.g. `basicSelectorTests`, `basicPropertyTests`, `basicAtRuleTests`, `basicValueTests`. It's then a little clear which object to reach for when creating a new rule.

**Comment by evilebottnawi** [comment]
@davidtheclark @jeddy3 `help wanted` and `feature`?

**Comment by jeddy3** [comment]
I'm kind of thinking these are a bit redundant as they'll basically be testing the `isStandardSyntax*` utils right? But these already have their own dedicated tests. It feels a bit like we're making work for ourselves.

Shall we close?

**Comment by davidtheclark** [comment]
I understand the point, though, which is to ensure that rules that *should* be using an `isStandardSyntax*` utils *actually are*.

**Comment by jeddy3** [comment]
> I understand the point, though, which is to ensure that rules that should be using an isStandardSyntax* utils actually are.

Oh, that makes more sense :)

Lets keep this issue open.

**Comment by jeddy3** [comment]
> We could create objects in testUtils/ that tests could import and use?

I've looked into this some more and I'm not sure it'll be this easy. I think the basic checks for `isStandardSyntax*` will, by their very nature, require the non-standard parsers to be used. I guess we could break `basicSelectorChecks` into `basicScssSelectorChecks` and `basicLessSelectorCheck`, but it's getting kind of bitty. Can anyone see a way around this, or is this approach acceptable?

**Comment by davidtheclark** [comment]
I don't think this issue is super important, so if it's too much of a pain let's just close it for now.

> isStandardSyntax* will, by their very nature, require the non-standard parsers to be used

I'm surprised. I thought SCSS syntax only breaks the parser with inline comments, and Less syntax with those and mixins?

**Comment by jeddy3** [comment]
> I thought SCSS syntax only breaks the parser with inline comments

[Seems to break](https://astexplorer.net/#/RHqac6LXB9) on interpolation too.

> so if it's too much of a pain let's just close it for now

Agreed.
